### PR TITLE
boards/samd21-based: Model features in Kconfig

### PR DESCRIPTION
**Pull request by jia200x**
### Contribution description
This PR models the features for samd21 based boards and adds them to the `Kconfig_features` test whitelist.

### Testing procedure

It should be enough to check that the `Kconfig_features` test works for all these boards (this means the list of provided features is equivalent):

```bash
BOARD=arduino-mkr1000 make && \
BOARD=arduino-mkrfox1200 make && \
BOARD=arduino-mkrwan1300 make && \
BOARD=arduino-mkrzero make && \
BOARD=arduino-zero make && \
BOARD=feather-m0 make && \
BOARD=hamilton make && \
BOARD=samd21-xpro make && \
BOARD=samr21-xpro make && \
BOARD=sensebox_samd21 make && \
BOARD=serpente make && \
BOARD=sodaq-autonomo make && \
BOARD=sodaq-explorer make && \
BOARD=sodaq-one make && \
BOARD=sodaq-sara-aff &&\
echo "OK"
```

### Issues/PRs references
~Depends on #13404~
#14148

### CODE REVIEW

File: boards/arduino-mkr1000/Kconfig [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Copyright (c) 2020 HAW Hamburg
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+source "$(RIOTBOARD)/common/arduino-mkr/Kconfig"
```
Comment (leandrolanzieri) [inline boards/arduino-mkr1000/Kconfig:7]: Looks like this file is missing?

File: boards/arduino-mkr1000/Kconfig [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+config BOARD
+    default "arduino-mkr1000" if BOARD_ARDUINO_MKR1000
+
+config BOARD_ARDUINO_MKR1000
+    bool
+    default y
+    select BOARDS_COMMON_ARDUINO-MKR
```
Comment (leandrolanzieri) [inline boards/arduino-mkr1000/Kconfig:15]: `-` can't be used in the symbol name
```suggestion
    select BOARDS_COMMON_ARDUINO_MKR
```

File: boards/sodaq-autonomo/Kconfig [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# Copyright (c) 2020 HAW Hamburg
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+source "$(RIOTBOARD)/common/sodaq/Kconfig"
```
Comment (leandrolanzieri) [inline boards/sodaq-autonomo/Kconfig:7]: This one seems to be missing as well

File: boards/arduino-mkr1000/Kconfig (hunk 1 above)
Comment (jia200x) [reply boards/arduino-mkr1000/Kconfig:7]: done

File: boards/sodaq-autonomo/Kconfig (hunk 3 above)
Comment (jia200x) [reply boards/sodaq-autonomo/Kconfig:7]: done

**Comment by leandrolanzieri** [comment]
This needs rebasing

File: boards/arduino-mkr1000/Kconfig [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+source "$(RIOTBOARD)/common/arduino-mkr/Kconfig"
+
+config BOARD
+    default "arduino-mkr1000" if BOARD_ARDUINO_MKR1000
+
+config BOARD_ARDUINO_MKR1000
+    bool
+    default y
+    select BOARDS_COMMON_ARDUINO_MKR
```
Comment (leandrolanzieri) [inline boards/arduino-mkr1000/Kconfig:15]: Here and bellow
```suggestion
    select BOARD_COMMON_ARDUINO_MKR
```

**Comment by leandrolanzieri** [comment]
This needs rebasing and adapting to the new classification

**Comment by leandrolanzieri** [comment]
@jia200x ping!

**Comment by leandrolanzieri** [comment]
@jia200x will not be having time so he said it's ok to take over. I rebased and applied some fixes.

@fjmolinas maybe you can take a look?

File: boards/sodaq-autonomo/Kconfig [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+config BOARD
+    default "sodaq-autonomo" if BOARD_SODAQ_AUTONOMO
+
+config BOARD_SODAQ_AUTONOMO
+    bool
+    default y
+    select CPU_MODEL_SAMD21J18A
+    select HAS_PERIPH_PWM
+    select HAS_ARDUINO_PWM
+    select BOARD_COMMON_SODAQ
```
Comment (aabadie) [inline boards/sodaq-autonomo/Kconfig:16]:
```suggestion
    select BOARD_COMMON_SODAQ
    select HAS_PERIPH_PWM
    select HAS_ARDUINO_PWM
```

**Comment by leandrolanzieri** [comment]
Rebased and addressed the comment!

File: boards/sodaq-sara-aff/Kconfig [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+# directory for more details.
+
+config BOARD
+    default "sodaq-sara-aff" if BOARD_SODAQ_SARA_AFF
+
+config BOARD_SODAQ_SARA_AFF
+    bool
+    default y
+    select CPU_MODEL_SAMD21J18A
+    select BOARD_COMMON_SODAQ
```
Comment (aabadie) [inline boards/sodaq-sara-aff/Kconfig:14]: In 05a5f3cedae96aa3b937bf7c499d9d71c76febe2 you moved the common symbol at the first place, for consistency, this should be the same here and with other sodaq boards.

```suggestion
    select BOARD_COMMON_SODAQ
    select CPU_MODEL_SAMD21J18A
```

File: boards/sodaq-one/Kconfig [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+# directory for more details.
+
+config BOARD
+    default "sodaq-one" if BOARD_SODAQ_ONE
+
+config BOARD_SODAQ_ONE
+    bool
+    default y
+    select CPU_MODEL_SAMD21G18A
+    select BOARD_COMMON_SODAQ
```
Comment (aabadie) [inline boards/sodaq-one/Kconfig:14]:
```suggestion
    select BOARD_COMMON_SODAQ
    select CPU_MODEL_SAMD21G18A
```

File: boards/sodaq-explorer/Kconfig [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+# directory for more details.
+
+config BOARD
+    default "sodaq-explorer" if BOARD_SODAQ_EXPLORER
+
+config BOARD_SODAQ_EXPLORER
+    bool
+    default y
+    select CPU_MODEL_SAMD21J18A
+    select BOARD_COMMON_SODAQ
```
Comment (aabadie) [inline boards/sodaq-explorer/Kconfig:14]:
```suggestion
    select BOARD_COMMON_SODAQ
    select CPU_MODEL_SAMD21J18A
```

**Comment by leandrolanzieri** [comment]
I just noticed that `sodaq-sara-sff` is missing

**Comment by aabadie** [comment]
> I just noticed that sodaq-sara-sff is missing

Indeed, it was added in the meantime...

**Comment by aabadie** [comment]
[wemos-zero](https://github.com/RIOT-OS/RIOT/tree/master/boards/wemos-zero) is not covered.

**Comment by leandrolanzieri** [comment]
> [wemos-zero](https://github.com/RIOT-OS/RIOT/tree/master/boards/wemos-zero) is not covered.

I'll add it

**Comment by leandrolanzieri** [comment]
I have added the `wemos-zero`, added the new `boards/common/arduino-zero` and modified `arduino-zero` to use this.

**Comment by leandrolanzieri** [comment]
Rebased once more :)

**Comment by aabadie** [comment]
> Rebased once more :)

Same here, I was also on it :)

**Pull request merged**